### PR TITLE
populate library with real metadata

### LINKED ISSUE #11: Populate music library with real metadata
I've fooled around with [musicmedata](https://www.npmjs.com/package/musicmetadata) and [fs-walk](https://www.npmjs.com/package/fs-walk) a bit but don't have anything concrete yet. @bcomnes let me know if you want to take a crack at this.

### PR DESCRIPTION
**Pull request by ungoldman**
> resolves #11

First pass at getting real metadata in there. It's scanning whatever directory is set in config every time a new window gets opened for now, but it's real data!

Next step is getting it to rescan when `state.config.music` changes. Then maybe caching metadata more intelligently.

### CODE REVIEW

File: renderer/lib/library.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const fs = require('fs')
+const path = require('path')
+const walker = require('folder-walker')
+const mm = require('musicmetadata')
+
+const extensions = ['.m4a', '.mp3']
+
+module.exports = (libPath, send, done) => {
```
Comment (ungoldman) [inline renderer/lib/library.js]: maybe i should replace `send, done` with a simple callback so this file isn't so tightly coupled with the model.

File: renderer/models/config.js [hunk 2]
```diff
@@ -1,30 +1,15 @@
...
+  return {
+    namespace: 'config',
+    state: config.store,
+    reducers: {
+      update: (data, state) => data
+    },
+    effects: {
+      set: (data, state, send, done) => {
+        config.set(data)
+        send('config:update', data, done)
```
Comment (ungoldman) [inline renderer/models/config.js]: not sure if setting state to `config.store` then calling `config.set` and updating state separately is the best way to do it. seems to work without problems for now but I foresee weirdness

**Comment by bcomnes** [comment]
LGTM!

**Comment by bcomnes** [comment]
screenshot 2016-08-09 10 14 26

**Comment by ungoldman** [comment]
@bcomnes I'm only using about 10 files to test now, guessing streaming every metadata update to state is too much.

**Comment by ungoldman** [comment]
I'm also running into a really weird bug where one file in particular causes electron to completely freeze without any console error or anything.

**Comment by bcomnes** [comment]
I imagine there are going to be worlds of edge cases to deal with in this problem.  Wrap in a setTimeout?

**Comment by ungoldman** [comment]
@bcomnes sounds good for now, I'll try that.

**Comment by ungoldman** [comment]
@bcomnes I tried putting `setTimeout` in a couple different places and didn't get great results. Gotta run for now, feel free to jump in.

**Comment by ungoldman** [comment]
Another option is to try throttling the stream -- either that or batching metadata updates.

**Comment by ungoldman** [comment]
I may have figured out what's causing a crash -- I think `musicmetadata` shoves the entire album art image in a buffer in the object it returns. I've got two files that are both causing a crash and they're the only two that have album art.

**Comment by ungoldman** [comment]
Okay gotta go for real, I might get to this today or might not. Need to get ready for a visit from my mom tomorrow followed by road trip to Seattle on Friday and flight to Paris Saturday! Busy week!

**Comment by bcomnes** [comment]
Np.  Nice work

**Comment by ungoldman** [comment]
Fixed the buffer issue and tweaked a couple minor things. Haven't gotten to try batching or throttling to slow down state updates from the metadata stream yet. Going to merge this for now as it's a good stopping point.

**Comment by bcomnes** [comment]
Read through the whole Choo README and started looking through your work. Looks great!  Go nate go!

**Pull request merged**